templates/common/on-prem/files/NetworkManager-resolv-prepender.yaml

```diff
@@ -81,33 +81,34 @@ contents:
             done
 
             # Image must exist before reaching this point otherwise timeout may cause image corruption.
-            NAMESERVER_IP=$(/usr/bin/podman run --rm \
+            NAMESERVER_IP="$(/usr/bin/podman run --rm \
                 --authfile /var/lib/kubelet/config.json \
                 --net=host \
                 {{ .Images.baremetalRuntimeCfgImage }} \
                 node-ip \
                 show \
                 --retry-on-failure \
                 {{range onPremPlatformAPIServerInternalIPs . }}"{{.}}" {{end}} \
-                {{range onPremPlatformIngressIPs . }}"{{.}}" {{end}})
+                {{range onPremPlatformIngressIPs . }}"{{.}}" {{end}})"
             DOMAINS="${IP4_DOMAINS} ${IP6_DOMAINS} {{.DNS.Spec.BaseDomain}}"
             if [[ -n "$NAMESERVER_IP" ]]; then
+                KNICONFDONEPATH="/run/resolv-prepender-kni-conf-done"
                 if systemctl -q is-enabled systemd-resolved; then
                     >&2 echo "NM resolv-prepender: Configure for OKD domain and local IP"
                     mkdir -p /etc/systemd/resolved.conf.d
                     KNICONFTMPPATH="$(mktemp)"
                     KNICONFPATH="/etc/systemd/resolved.conf.d/60-kni.conf"
-                    KNICONFDONEPATH="/run/resolv-prepender-kni-conf-done"
                     echo "[Resolve]" > "${KNICONFTMPPATH}"
                     echo "DNS=$NAMESERVER_IP" >> "${KNICONFTMPPATH}"
-                    echo "Domains=$DOMAINS" >> "${KNICONFTMPPATH}"
+                    echo "Domains=${DOMAINS}" >> "${KNICONFTMPPATH}"
 
                     # If KNI conf is not created or doesn't match what is generated or
                     # if we haven't completed a full update - create or update it
                     if [[ ! -f "${KNICONFPATH}" ]] || [[ ! -f "${KNICONFDONEPATH}" ]] || ! cmp --silent "${KNICONFPATH}" "${KNICONFTMPPATH}"; then
                         >&2 echo "NM resolv-prepender: Creating/updating /etc/systemd/resolved.conf.d/60-kni.conf"
-                        rm -f "${KNICONFDONEPATH}"
```

Comment:
This was left here in case an interrupt occurred between writing to the file and restarting systemd-resolved so that in a following iteration, it would create the kni conf and restart the systemd service.

Reply:
Ah, I see, let me revert this part